src/components/StatePicker/StateSelectorModal.tsx

```diff
@@ -1,48 +1,41 @@
 import {CONST as COMMON_CONST} from 'expensify-common/lib/CONST';
-import PropTypes from 'prop-types';
 import React, {useEffect, useMemo} from 'react';
-import _ from 'underscore';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import Modal from '@components/Modal';
 import ScreenWrapper from '@components/ScreenWrapper';
 import SelectionList from '@components/SelectionList';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
-import searchCountryOptions from '@libs/searchCountryOptions';
+import searchCountryOptions, {type CountryData} from '@libs/searchCountryOptions';
 import StringUtils from '@libs/StringUtils';
 import CONST from '@src/CONST';
 
-const propTypes = {
+type State = keyof typeof COMMON_CONST.STATES;
+
+type StateSelectorModalProps = {
     /** Whether the modal is visible */
-    isVisible: PropTypes.bool.isRequired,
+    isVisible: boolean;
 
     /** State value selected  */
-    currentState: PropTypes.string,
+    currentState?: State | '';
```

Comment:
wondering if we need to pass this `''` to this type? and in default props just pass `undefined`, wdyt?